common/cards/alter-egos-iii/hermits/architectfalse-rare.ts

```diff
@@ -37,36 +49,51 @@ class ArchitectFalseRare extends Card {
 		},
 	}
 
-	lastAttacker = new InstancedValue<CardComponent | null>(() => null)
-	lastAttackType = new InstancedValue<HermitAttackType | null>(() => null)
+	lastAttackInfo = new GameValue<Record<PlayerEntity, AttackInfo | undefined>>(() => {
+		return {}
+	})
 
 	override onCreate(game: GameModel, component: CardComponent) {
-		const {opponentPlayer} = component
+		if (Object.hasOwn(this.lastAttackInfo, game.id)) return
```

Comment:
Correct me if im wrong, but I believe this will always be false?

Reply:
Most decks will have multiple copies of the same hermit, this should reduce allocating duplicate resources and prevent data loss if another GA CardComponent is created in the middle of the game

Reply:
I believe `Object.hasOwn` is checking the GameValue object and not the instance at the moment. i think for that intention this line needs to be `if (Object.hasOwn(this.lastAttackInfo.values, game.id))`.